Back-To-School/Assets/BackToSchool/Scripts/Inputs/PlayerInputProvider.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using Assets.BackToSchool.Scripts.Interfaces.Input;
+using UnityEngine;
+
+
+namespace Assets.BackToSchool.Scripts.Inputs
+{
+    public class PlayerInputProvider : BaseInput, IPlayerInput
+    {
+        public event Action<Vector3> Moved;
+        public event Action<Vector3> Rotated;
+        public event Action Stopped;
+        public event Action Fired;
+        public event Action Reloaded;
+        public event Action<bool> WeaponChanged;
+
+        private readonly Camera _mainCamera;
+        private RaycastHit _previousHit;
+        private RaycastHit _hit;
+        private Ray _ray;
+        private LayerMask _layerMask = LayerMask.GetMask("Ground");
+
+        private float _rayCastLength = 100f;
+        private bool _isPaused;
+
+        public PlayerInputProvider(Camera mainCamera) => _mainCamera = mainCamera;
+
+        public void SetIsPause(bool isPause) => _isPaused = isPause;
```

Comment:
I'd suggest to do it as a part of IInputProvider and place in BaseInput.
Also, please rename it to `TogglePause(isPaused)` or split it to 2 methods: Pause and Continue